docs/run_notebooks.py

```diff
@@ -0,0 +1,196 @@
+import io
+import os
+import json
+import time
+import pathlib
+import argparse
+import warnings
+import traceback
+from shutil import copyfile
+from contextlib import redirect_stderr, redirect_stdout
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+
+
+def execute_notebook(nb_test_dir, filename):
+    os.environ["JUPYTER_PLATFORM_DIRS"] = "1"
+
+    with open(filename) as fp:
+        nb = json.load(fp)
+
+    mpl.use("agg")  # Use a non-interactive backend
+    os.chdir(nb_test_dir)
```

Comment:
changing the current working directory in scripts it's usually discouraged, but I understand the need here
as a best practice, let's reset the previous CWD at the end of the function

Reply:
Good point! I'll add a little context manager to do this. Then it also doesn't clutter up the code too much.